ipc/rpc/src/binary.rs

```diff
@@ -154,6 +171,8 @@ impl<T> BinaryConvertable for Vec<T> where T: BinaryConvertable {
 				_ => 128,
 			});
 
+		if buffer.len() == 0 { return Ok(result); }
```

Comment:
since this code can only be reached if accessing `buffer[0]` didn't panic, it is unreachable. probably should be moved above it